packages/web-components/src/scripts/build-prod.js

```diff
@@ -1,50 +1,107 @@
+const tsDeclarationIndexTemplateFilePath = path.resolve(__dirname, './tpls/index.d.ts.ejs')
+
+const generateFileForPackages = async ({ packages, tplPath, destinationFilePath, destinationFolderPath }) => {
+  const compileResult = await ejs.renderFile(tplPath, {
+    packages,
+  })
+
+  const formatResult = formatCode(compileResult)
+
+  await mkdirp(destinationFolderPath)
+  console.log(`created ${destinationFilePath}`)
+  await writeFileSync(destinationFilePath, formatResult, { flag: 'w' })
 }
 
-return (() => {
-  const { execSync } = require('child_process')
-  const { setEnv } = require('./get-env')
-  const packages = ['search-widget', 'appointment-bookings', 'viewing-booking', 'themes']
+return (async () => {
+  const packages = [
+    {
+      webpackPackageName: 'search-widget',
+      exportName: 'ReapitSearchWidgetComponent',
+    },
+    {
+      webpackPackageName: 'appointment-bookings',
+      exportName: 'ReapitAppointmentBookingComponent',
+    },
+    {
+      webpackPackageName: 'viewing-booking',
+      exportName: 'ReapitViewingBookingComponent',
+    },
+    {
+      webpackPackageName: 'themes',
+    },
+  ]
   const opts = {
     stdio: 'inherit',
   }
   setEnv()
 
-  const clearPublic = 'rimraf ./public/dist && rimraf ./public/themes'
+  const clearPublic = 'rimraf ./public/dist-npm && rimraf ./public/dist && rimraf ./public/themes'
   execSync(clearPublic, opts)
 
-  packages.forEach(package => {
-    try {
-      const clientScript = `rollup -c './src/scripts/rollup.config.${package}.js'`
-      execSync(clientScript, opts)
-    } catch (err) {
-      console.error('Error in dev server', err)
-      process.exit(1)
-    }
+  const promises = packages.map(({ webpackPackageName }) => {
+    const buildPackageFn = () =>
+      new Promise((resolve, reject) => {
+        {
+          const clientBuildScriptPath = `./src/scripts/rollup.config.${webpackPackageName}.js`
+          const spawnObject = spawn('rollup', ['-c', clientBuildScriptPath], {
+            stdio: 'inherit',
+          })
+          spawnObject.on('error', err => reject(err))
+          spawnObject.on('exit', resolve)
+        }
+      })
+
+    return buildPackageFn()
+  })
+
+  promises.push(
+    generateFileForPackages({
+      packages,
+      tplPath: cjsIndexTemplateFilePath,
+      destinationFilePath: cjsIndexFilePath,
+      destinationFolderPath: cjsIndexFolderPath,
+    }),
+    generateFileForPackages({
+      packages,
+      tplPath: esmIndexTemplateFilePath,
+      destinationFilePath: esmIndexFilePath,
+      destinationFolderPath: esmIndexFolderPath,
+    }),
+    generateFileForPackages({
+      packages,
+      tplPath: tsDeclarationIndexTemplateFilePath,
+      destinationFilePath: tsDeclarationIndexFilePath,
+      destinationFolderPath: tsDeclarationIndexFolderPath,
+    }),
+  )
+
+  await Promise.all(promises).catch(err => {
+    console.error('There is an error while building: ')
+    console.error(err)
+    process.exit(1)
```

Comment:
This looks weird to me, can we use `try/catch` with `await` instead of `await catch` ?